interface/modules/custom_modules/oe-module-faxsms/src/Controller/RCFaxClient.php

```diff
@@ -0,0 +1,1231 @@
+    {
+        $authErrorMsg = $this->authenticate();
+        if ($authErrorMsg !== 1) {
+            return text(js_escape($authErrorMsg)); // goes to alert
+        }
+
+        try {
+            $response = $this->platform->get("/account/~/extension/~/message-store/{$messageId}/content");
+            $contentType = $response->response()->getHeader('Content-Type')[0];
+            $fileExtension = $this->getFileExtension($contentType);
+            $fileName = "fax_{$messageId}." . $fileExtension;
+
+            // Save the file locally
+            $filePath = $this->cacheDir . DIRECTORY_SEPARATOR . $fileName;
+            file_put_contents($filePath, $response->raw());
+
+            // Prepare the file for download
+            header('Content-Description: File Transfer');
+            header('Content-Type: ' . $contentType);
+            header('Content-Disposition: attachment; filename="' . basename($filePath) . '"');
+            header('Content-Transfer-Encoding: binary');
+            header('Expires: 0');
+            header('Cache-Control: must-revalidate');
+            header('Pragma: public');
+            header('Content-Length: ' . filesize($filePath));
+            readfile($filePath);
+
+            // Optionally, you can delete the file after download
+            unlink($filePath);
+
+            exit; // Stop further script execution
+        } catch (ApiException $e) {
+            return text(json_encode(['error' => "API Error: " . $e->getMessage()]));
+        } catch (Exception $e) {
+            return text(json_encode(['error' => "Error: " . $e->getMessage()]));
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getUser(): string
+    {
+        $id = $this->getRequest('uid');
+        $query = "SELECT * FROM users WHERE id = ?";
+        $result = sqlStatement($query, [$id]);
+        $u = sqlFetchArray($result);
+        return json_encode([$u['fname'], $u['lname'], $u['fax']]);
+    }
+
+    /**
+     * @return string
+     */
+    public function getNotificationLog(): string
+    {
+        $type = $this->getRequest('type');
+        $fromDate = $this->getRequest('datefrom');
+        $toDate = $this->getRequest('dateto');
+        try {
+            $query = "SELECT notification_log.* FROM notification_log WHERE notification_log.dSentDateTime > ? AND notification_log.dSentDateTime < ?";
+            $res = sqlStatement($query, [$fromDate, $toDate]);
+            $responseMsg = '';
+            while ($nrow = sqlFetchArray($res)) {
+                $adate = ($nrow['pc_eventDate'] . '::' . $nrow['pc_startTime']);
+                $pinfo = str_replace("|||", " ", $nrow['patient_info']);
+                $msg = htmlspecialchars($nrow["message"], ENT_QUOTES);
+                $responseMsg .= "<tr><td>" . $nrow["pc_eid"] . "</td><td>" . $nrow["dSentDateTime"] . "</td><td>" . $adate . "</td><td>" . $pinfo . "</td><td>" . $msg . "</td></tr>";
```

Comment:
looks like need text() around $nrow["pc_eid"], $nrow["dSentDateTime"], $adate, and $pinfo